### PR TITLE
Cleanup build and packages

### PR DESCRIPTION
**Pull request by darkdragon-001**
- Improve Flatpak, Debian and AppImage. Didn't touch Clickable and Snap
- More (re-)use of `Makefile` targets
- (Re-)use meta data among package formats
- Cleanup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,49 +48,8 @@ To find out how to build from source and install yourself, please see below.
...
-    `lorca`-> native chromium (has to be installed),
-    `ut` -> runs in the ut enviroment,
-    `me` -> qmlscene,
-    `server` -> just run the webserver. Defaults to run with `electron`.
-* `-eDebug` show developer console in electron mode
-* `-version` Print version info
-* `-host` Set the host to run the webserver from. Defaults to localhost.
-* `-port` Set the port to run the webserver from. Defaults to 9080.
-
-## Environment variables
```
Comment (nanu-c) [inline README.md:90]: it still has run flags, we need to update them and not only delete it.

File: Makefile [hunk 2]
```diff
@@ -131,215 +131,12 @@ debug-flatpak:
...
-build-deb-arm64: clean-deb-arm64
-	@echo "Building Axolotl for arm64/aarch64 on Debian - Please use 'testing' release!)."
-	@echo "Installing dependencies (yarn)..."
-	@cd $(CURRENT_DIR)/axolotl-web && yarn install --frozen-lockfile
-	@echo "Building (yarn)..."
-	@cd $(CURRENT_DIR)/axolotl-web && yarn run build
-	@echo "Building (rust)..."
-	$(CARGO_PREFIX)/cargo build --features tauri --release
-	@echo "Building complete."
-
```
Comment (nanu-c) [inline Makefile:180]: How is the deb package now built? `cargo tauri build --ci --target aarch64-unknown-linux-gnu --features tauri --bundles deb` -> does it work?

**Review by nanu-c — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (nanu-c) [reply README.md:90]: https://github.com/nanu-c/axolotl/blob/c248f457ec1372d9cfa1c8f5d91e095d969f4f2f/src/main.rs#L9-L15

Comment (darkdragon-001) [reply README.md]: Done.

File: Makefile (hunk 2 above)
Comment (darkdragon-001) [reply Makefile:180]: `--ci` is only needed for non-interactive builds.
`--target aarch64-unknown-linux-gnu` is only needed for cross-compilation.

So if you want to build for your current arch, it is only `cargo tauri build --features tauri --bundles deb`.

File: Makefile [hunk 3]
```diff
@@ -33,72 +33,65 @@ WGET := $(shell which wget 2>/dev/null)
...
 CROSS := $(shell which cross 2>/dev/null)
 DOCKER := $(shell which docker 2>/dev/null)
-ASTILECTRON_BUILDER := $(shell which astilectron-bundler 2>/dev/null)
 
-DESTDIR := /
-INSTALL_PREFIX := usr/bin
-LIBRARY_PREFIX := usr/lib
-SHARE_PREFIX := usr/share
+INSTALL_PREFIX := "/usr"
+bindir := $(INSTALL_PREFIX)/bin
```
Comment (olof-nord) [inline Makefile:38]: Question: I notice this is lowercased - what does that mean? Asking to understand better

File: docs/INSTALL.md [hunk 4]
```diff
@@ -120,179 +119,81 @@ To install the built snap, use snap:
...
+`flatpak run org.nanuc.Axolotl --mode tauri`
 
 ## AppImage
 
-**Tooling**
+### Tooling
 
-This requires `appimagetool`, `go` and `npm` to be installed locally.
-Installation instructions for `appimagetool` can be found [here](https://github.com/AppImage/AppImageKit#appimagetool-usage)
+This requires `yarn`, `cargo` and `tari-cli` to be installed locally.
```
Comment (olof-nord) [inline docs/INSTALL.md:162]: Typo? tauri-cli

File: flatpak/org.nanuc.Axolotl.yml [hunk 5]
```diff
@@ -88,3 +58,38 @@ modules:
           --install.offline true
           --run.offline true
         dest-filename: .yarnrc
+    modules:
+      - name: abseil
```
Comment (olof-nord) [inline flatpak/org.nanuc.Axolotl.yml:62]: What does abseil do?

**Review by olof-nord — COMMENTED** [review]
Huge! Impressive work

File: flatpak/org.nanuc.Axolotl.yml (hunk 5 above)
Comment (nanu-c) [reply flatpak/org.nanuc.Axolotl.yml:62]: I think abseil is a dependency of protoc to compile the protobufs

File: Makefile (hunk 3 above)
Comment (darkdragon-001) [reply Makefile:38]: I don't have a strong opinion. I noticed in the [GNU Make manual](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html) they use these variable names. In [CMake](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) they use capital ones. I think it is advisable to use standard names but I don't have a strong opinion on the casing.

File: flatpak/org.nanuc.Axolotl.yml (hunk 5 above)
Comment (darkdragon-001) [reply flatpak/org.nanuc.Axolotl.yml:62]: Exactly. Previously, we used `git` which also downloaded Git submodules. As we now download via HTTP, we have to download the dependencies separately.

File: Makefile (hunk 3 above)
Comment (darkdragon-001) [reply Makefile]: I changed to using capital letters.

**Comment by darkdragon-001** [comment]
I resolved all issues. @nanu-c could you please take another look?

**Review by nanu-c — APPROVED** [review]

**Pull request merged**